centos2ol.sh

```diff
@@ -220,345 +290,384 @@ You may want select No to stop and raise an issue on ${github_url} for advice."
+fi
+step_info $LINENO "Creating temporary directory for files"
+if (( STEPNO >= STARTNO  )) ; then
+   #cd "$(mktemp -d)"
+   mkdir $BASE/temp
+fi
+   
+cd $BASE/temp
 
+if (( STEPNO >= STARTNO  )) ; then
 # Most distros keep their /etc/yum.repos.d content in the -release rpm. CentOS 8 does not and the behaviour changes between
 #  minor releases; 8.0 uses 'centos-repos' while 8.3 uses 'centos-linux-repos', glob for simplicity.
+
 if [[ $old_release =~ ^centos-release-8.* ]] || [[ $old_release =~ ^centos-linux-release-8.* ]]; then
     old_release=$(rpm -qa centos*repos)
 fi
-
-echo "Backing up and removing old repository files..."
-# Identify repo files from the base OS
-rpm -ql "$old_release" | grep '\.repo$' > repo_files
-# Identify repo files from 'CentOS extras'
-if [ "$(rpm -qa "centos-release-*" | wc -l)" -gt 0 ] ; then
-    rpm -qla "centos-release-*" | grep '\.repo$' >> repo_files
 fi
-while read -r repo; do
-    if [ -f "$repo" ]; then
-        cat - "$repo" > "$repo".disabled <<EOF
+
+step_info $LINENO "Backing up and removing old repository files..."
+# Identify repo files from the base OS and write them to $BASE/temp/repo_files
+if (( STEPNO >= STARTNO  )) ; then
+   rpm -ql "$old_release" | grep '\.repo$' > repo_files
+   # Identify repo files from 'CentOS extras'
+   if [ "$(rpm -qa "centos-release-*" | wc -l)" -gt 0 ] ; then
+       rpm -qla "centos-release-*" | grep '\.repo$' >> repo_files
+   fi
+   while read -r repo; do
+       if [ -f "$repo" ]; then
+           cat - "$repo" > "$repo".disabled <<EOF
 # This is a yum repository file that was disabled by
 # ${0##*/}, a script to convert CentOS to Oracle Linux.
 # Please see $yum_url for more information.
 
 EOF
-        tmpfile=$(mktemp repo.XXXXX)
-        echo "$repo" | cat - "$repo" > "$tmpfile"
-        rm "$repo"
-    fi
-done < repo_files
-
-echo "Downloading Oracle Linux release package..."
-if ! yumdownloader "${new_releases[@]}"; then
-    {
-        echo "Could not download the following packages from $yum_url:"
-        echo "${new_releases[@]}"
-        echo
-        echo "Are you behind a proxy? If so, make sure the 'http_proxy' environment"
-        echo "variable is set with your proxy address."
-    } >&2
-    restore_repos
+           tmpfile=$(mktemp repo.XXXXX)
+           echo "$repo" | cat - "$repo" > "$tmpfile"
+           rm "$repo"
+       fi
+   done < repo_files
 fi
 
-echo "Switching old release package with Oracle Linux..."
-rpm -i --force "${new_releases[@]/%/*.rpm}"
-rpm -e --nodeps "$old_release"
-rm -f "${reposdir}/switch-to-oraclelinux.repo"
+step_info $LINENO "Downloading Oracle Linux release package..."
+if (( STEPNO >= STARTNO  )) ; then
+   trap error_handler1 ERR
+   if ! yumdownloader "${new_releases[@]}"; then
+       {
+           echo "Could not download the following packages from $yum_url:"
+           echo "${new_releases[@]}"
+           echo
+           echo "Are you behind a proxy? If so, make sure the 'http_proxy' environment"
```

Comment:
This should be `https_proxy` and I'll flag that for the current version as well.